src/token.sol

```diff
@@ -17,86 +17,120 @@
 
 pragma solidity >=0.4.23;
 
-import "ds-stop/stop.sol";
+import "ds-math/math.sol";
+import "ds-auth/auth.sol";
 
-import "./base.sol";
 
-contract DSToken is DSTokenBase(0), DSStop {
-
-    bytes32  public  symbol;
-    uint256  public  decimals = 18; // standard token precision. override to customize
+contract DSToken is DSMath, DSAuth {
+    bool                                              public  stopped;
+    uint256                                           public  totalSupply;
+    mapping (address => uint256)                      public  balanceOf;
+    mapping (address => mapping (address => uint256)) public  allowance;
+    bytes32                                           public  symbol;
+    uint256                                           public  decimals = 18; // standard token precision. override to customize
+    bytes32                                           public  name = "";     // Optional token name
 
     constructor(bytes32 symbol_) public {
         symbol = symbol_;
     }
 
+    event Approval(address indexed src, address indexed guy, uint wad);
+    event Transfer(address indexed src, address indexed dst, uint wad);
     event Mint(address indexed guy, uint wad);
     event Burn(address indexed guy, uint wad);
+    event Stop();
+    event Start();
+
+    modifier stoppable {
+        require(!stopped, "ds-stop-is-stopped");
+        _;
+    }
 
-    function approve(address guy) public stoppable returns (bool) {
-        return super.approve(guy, uint(-1));
+    function approve(address guy) external returns (bool) {
```

Comment:
why not keep this as `stoppable`?

Reply:
the `approve` that it calls is stoppable anyway. But yeah, making it explicit is nicer I suppose

Reply:
ah of course, missed that :woman_facepalming: 

I'm fine with leaving this as it is then, it's actually more consistent wtih the other methods that call into `stoppable` functions (`push`, `pull`, `move`, etc).